src/Tuura/SelfDual.hs

```diff
@@ -0,0 +1,56 @@
+module Tuura.SelfDual (
+    isSelfDual, getSelfDuals) where
+
+import Data.List (sort)
+import Control.Monad (replicateM)
+
+import Tuura.Boolean
```

Comment:
Aha, it looks like `Tuura.Boolean` deserves to be a separate library, so everyone can reuse it without changing Plato :-)